CONTRIBUTING.md

```diff
@@ -0,0 +1,30 @@
+# How to successfully make changes to Ninja
+
+We're very wary of changes that increase the complexity of Ninja (in particular,
+new build file syntax or command-line flags) or increase the maintenance burden
+of Ninja. Ninja is already successfully used by hundreds of developers for large
+projects and it already achieves (most of) the goals we set out for it to do.
+It's probably best to discuss new feature ideas on the
+[mailing list](https://groups.google.com/forum/#!forum/ninja-build) or in an
+issue before creating a PR.
```

Comment:
Very little traffic is a plus for many, as they don't want to read through people asking questions on how to use Ninja on GitHub and other noise.

> such as inline reviews associated with the code.

The discussion should happen before there is any code.

Reply:
There's also the issue of Google Groups allowing non google accounts to participate. Is that even possible? Maybe it is and I'm just mis-informed.

Also: What discussion? Most of the architectural discussion for changes to Ninja already happens here on github. The *overwhelming majority* of Ninja related discussion happens here on github. A mailing list that is never used by anyone, even the maintainers of the project, is harmful to the project, because it leads potential contributors to believe that they *should* post to the mailing list, but when they do, they are ignored or get no feedback for several months. All the while the maintainers of the project have multi dozen paragraph technical discussions on github.

Either forbid anything but line-by-line code-review on github, and force all architectural discussion to be on the mailing list, or drop the mailing list. All this is doing is splitting the community into two discussion venues.